guvnor-structure/guvnor-structure-client/src/main/java/org/guvnor/structure/client/editors/repository/clone/CloneRepositoryPresenter.java

```diff
@@ -0,0 +1,211 @@
+        }
+    }
+
+    private void populateOrganizationalUnits() {
+        //populate Organizational Units list box
+        organizationalUnitService.call(
+            new RemoteCallback<Collection<OrganizationalUnit>>() {
+                @Override
+                public void callback(Collection<OrganizationalUnit> organizationalUnits) {
+                    view.addOrganizationalUnitSelectEntry();
+                    if (organizationalUnits != null && !organizationalUnits.isEmpty()) {
+                        for (OrganizationalUnit organizationalUnit : organizationalUnits) {
+                            view.addOrganizationalUnit(organizationalUnit.getName(),
+                                    organizationalUnit.getName());
+                            availableOrganizationalUnits.put(organizationalUnit.getName(),
+                                    organizationalUnit);
+                        }
+                    }
+                }
+            },
+            new ErrorCallback<Message>() {
+                @Override
+                public boolean error(final Message message,
+                                     final Throwable throwable) {
+                    view.errorLoadOrganizationalUnitsFail(throwable);
+                    return false;
+                }
+            }
+        ).getOrganizationalUnits();
+    }
+
+    public void showForm() {
+        view.showForm();
+    }
+
+    private void lockScreen() {
+        view.showBusyPopupMessage();
+        view.setPopupCloseVisible( false );
+        view.setCloneEnabled( false );
+        view.setCancelEnabled( false );
+        view.setPasswordEnabled( false );
+        view.setUsernameEnabled( false );
+        view.setGitUrlEnabled( false );
+        view.setOrganizationalUnitEnabled( false );
+        view.setNameEnabled(false);
+    }
+
+    private void unlockScreen() {
+        view.closeBusyPopup();
+        view.setPopupCloseVisible(true);
+        view.setCloneEnabled(true);
+        view.setCancelEnabled(true);
+        view.setPasswordEnabled(true);
+        view.setUsernameEnabled(true);
+        view.setGitUrlEnabled(true);
+        view.setOrganizationalUnitEnabled( true );
+        view.setNameEnabled(true);
+    }
+
+    protected boolean isOuMandatory() {
+        try {
+            final IOCBeanDef<RepositoryPreferences> beanDef = IOC.getBeanManager().lookupBean(RepositoryPreferences.class);
+            return beanDef == null || beanDef.getInstance().isOUMandatory();
+        } catch ( IOCResolutionException exception ) {
+        }
+        return true;
+    }
+
+    protected void setRepositoryService(Caller<RepositoryService> repositoryService) {
+        this.repositoryService = repositoryService;
+    }
+
+    protected void setOrganizationalUnitService(Caller<OrganizationalUnitService> organizationalUnitService) {
+        this.organizationalUnitService = organizationalUnitService;
+    }
```

Comment:
Sure. Doesn't have to be constructor injection if you figure out another way. The goal is just to keep code that is used for testing on the test side and out of the main side.